From 2023e36a7f082ba449d10cbf4850b4e0258adb55 Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Fri, 17 Oct 2014 16:42:29 +0800
Subject: [PATCH] driver: flash: update function prototype

As the load function re-defined, so update the flash load
function accordingly.

Signed-off-by: Bo Shen <voice.shen@atmel.com>
---
 driver/flash.c  | 24 ++++++++++++++----------
 include/flash.h |  3 +--
 2 files changed, 15 insertions(+), 12 deletions(-)

diff --git a/driver/flash.c b/driver/flash.c
index ebc5d5a..714c922 100644
--- a/driver/flash.c
+++ b/driver/flash.c
@@ -22,17 +22,21 @@
  * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
  * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
  * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-*/
-#include "../include/part.h"
-#include "../include/main.h"
-#include "../include/flash.h"
+ */
 
-int load_norflash(unsigned int img_addr,
-		unsigned int img_size,
-		unsigned int img_dest)
+#include "common.h"
+#include "flash.h"
+#include "string.h"
+#include "debug.h"
+
+int load_norflash(struct image_info *image)
 {
-    norflash_hw_init();
+	norflash_hw_init();
+
+	dbg_info("FLASH: copy %d bytes from %x to %x\n",
+		 image->length, image->offset, image->dest);
+
+	memcpy(image->dest, (const char *)image->offset, image->length);
 
-    memcpy((char *)img_dest, (char *)(AT91_NORFLASH_BASE + img_addr), img_size);
-    return 0;
+	return 0;
 }
diff --git a/include/flash.h b/include/flash.h
index d9e5dc4..bd5c528 100644
--- a/include/flash.h
+++ b/include/flash.h
@@ -44,7 +44,6 @@
 
 void norflash_hw_init(void);
 
-int load_norflash(unsigned int img_addr, unsigned int img_size,
-                  unsigned int img_dest);
+int load_norflash(struct image_info *image);
 
 #endif	/* #ifndef __NORFLASH_H__ */
-- 
1.9.1

